--- ../src-base/minecraft/net/minecraft/item/ItemEndCrystal.java
+++ ../src-work/minecraft/net/minecraft/item/ItemEndCrystal.java
@@ -1,17 +1,27 @@
 package net.minecraft.item;
 
 import java.util.List;
+
+import carpet.CarpetSettings;
+import hippogriff.logger.ShotaWarn;
+import hippogriff.text.TextBuilder;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.creativetab.CreativeTabs;
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.item.EntityEnderCrystal;
 import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.entity.player.EntityPlayerMP;
 import net.minecraft.init.Blocks;
+import net.minecraft.network.play.server.SPacketChat;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.server.management.PlayerList;
 import net.minecraft.util.EnumActionResult;
 import net.minecraft.util.EnumFacing;
 import net.minecraft.util.EnumHand;
 import net.minecraft.util.math.AxisAlignedBB;
 import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.text.TextComponentString;
+import net.minecraft.util.text.TextFormatting;
 import net.minecraft.world.World;
 import net.minecraft.world.WorldProviderEnd;
 import net.minecraft.world.end.DragonFightManager;
@@ -66,6 +76,10 @@
                     {
                         if (!p_180614_2_.field_72995_K)
                         {
+                            //Hippogriff SMP: Log EnderCrystals
+                            if (p_180614_1_.func_184102_h() != null && CarpetSettings.warnLogs)
+                                ShotaWarn.warn((EntityPlayerMP) p_180614_1_, p_180614_3_.toString(), "ha colocado un EnderCrystal en", "has placed an EnderCrystal in");
+
                             EntityEnderCrystal entityendercrystal = new EntityEnderCrystal(p_180614_2_, (double)((float)p_180614_3_.func_177958_n() + 0.5F), (double)(p_180614_3_.func_177956_o() + 1), (double)((float)p_180614_3_.func_177952_p() + 0.5F));
                             entityendercrystal.func_184517_a(false);
                             p_180614_2_.func_72838_d(entityendercrystal);
